src/lib/server/subscriptionUtils.ts

```diff
@@ -94,12 +94,13 @@ export const createStripeSession = async (
   email: string,
   name: string,
   origin: string,
-  options?: { successUrl?: string; subscriptionType?: 'yearly' | 'monthly' },
+  options?: { successUrl?: string; subscriptionType?: 'yearly' | 'monthly'; promoCode?: string },
 ) => {
   const priceCode = options?.subscriptionType === 'yearly' ? YEARLY_PRICE_CODE : MONTHLY_PRICE_CODE;
   console.log(`Fetching Stripe customer ID for user ${uid}`);
   const stripeCustomerIdRef = firebaseDb.ref(`/users/${uid}/private/stripeCustomerId`);
 
+  const promoCodesPromise = options?.promoCode ? stripeClient.promotionCodes.list({ code: options.promoCode }) : Promise.resolve({ data: [] });
```

Comment:
Somewhat obnoxious but unfortunately Stripe doesn't seem to let me pass a string as a promo code and just try it, same as if a user entered an invalid one. It errors out if it is invalid. So instead we need to check if it exists and apply it. I do the promise up here and await it below for max parallelization.